### PR TITLE
add helper OnSysCall for use in blockchain toolkit

### PR DESCRIPTION
**Pull request by devhawk**
Add protected OnSysCall overload that can be called by blockchain toolkit libraries

### CODE REVIEW

**Review by shargon — APPROVED** [review]
good change array instead of list

**Comment by superboyiii** [comment]
@erikzhang Need your review.

File: src/neo/SmartContract/ApplicationEngine.cs [hunk 1]
```diff
@@ -302,19 +302,31 @@ protected void ValidateCallFlags(CallFlags requiredCallFlags)
                 throw new InvalidOperationException($"Cannot call this SYSCALL with the flag {state.CallFlags}.");
         }
 
-        protected override void OnSysCall(uint method)
+        protected void OnSysCall(InteropDescriptor descriptor, Func<object, object[], object> handler)
```
Comment (erikzhang) [inline src/neo/SmartContract/ApplicationEngine.cs:305]: Why `Func<object, object[], object>` instead of `descriptor.Handler`?

Comment (devhawk) [reply src/neo/SmartContract/ApplicationEngine.cs:305]: So the debugger can override the behavior of existing interop services such as `System.Runtime.CheckWitness`

Example: https://github.com/ngdenterprise/neo-blockchaintoolkit-library/blob/b16214cfc059a4175311dbbb3bd67a90b4ecd144/src/bctklib-3/smart-contract/TestApplicationEngine.cs#L80

**Review by superboyiii — APPROVED** [review]

File: src/neo/SmartContract/ApplicationEngine.cs (hunk 1 above)
Comment (erikzhang) [reply src/neo/SmartContract/ApplicationEngine.cs:305]: Please check #2382. I think it can solve this problem.

Comment (devhawk) [reply src/neo/SmartContract/ApplicationEngine.cs:305]: I can make #2382 work

**Pull request closed without merging**